src/plugins/cft/api.js

```diff
@@ -0,0 +1,244 @@
+
+    return response.body.data;
+}
+
+/**
+ * @returns {Promise.<Array>}
+ */
+async function fetchWallets() {
+    const response = await fetchJson(makeApiUrl("wallets"), {
+        log:                 logFetches,
+        headers:             defaultHeaders,
+        sanitizeRequestLog:  {},
+        sanitizeResponseLog: {},
+    });
+
+    assertResponseSuccess(response);
+
+    return response.body.hasOwnProperty('data') && response.body.data.hasOwnProperty('wallets')
+        ? response.body.data.wallets
+        : [];
+}
+
+/**
+ * @param dateFrom
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactions(dateFrom) {
+    let isFirstPage  = true;
+    let transactions = [];
+
+    const lastSyncTime = new Date(dateFrom).getTime();
+    const pagination   = {
+        limit:  20,
+        offset: 0,
+        total:  null,
+    };
+
+    console.log('last sync: ' + lastSyncTime + ' (' + dateFrom + ')');
+
+    while (isFirstPage || pagination.offset < pagination.total) {
+        const data = await fetchTransactionsInternal(pagination.limit, pagination.offset);
+
+        if (data.part.offset !== pagination.offset) { // банк ограничивает выборку
+            break;
+        }
+
+        if (isFirstPage) {
+            pagination.total = data.part.totalCount;
+            isFirstPage      = false;
+        }
+
+        const loadNext = data.data
+                             .filter((transaction) => transaction.itemType === 'OPERATION' && ('money' in transaction))
+                             .every((transaction) => {
+                                 const isActual = transaction.date > lastSyncTime;
+
+                                 isActual && transactions.push(transaction);
+
+                                 return isActual;
+                             });
+
+        if (loadNext) {
+            pagination.offset += pagination.limit;
+        } else {
+            break;
+        }
+    }
+
+    return transactions;
+}
+
+/**
+ * @param limit
+ * @param offset
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactionsInternal(limit, offset) {
+    const urlParams = {
+        'limit':  limit,
+        'offset': offset,
+    };
+
+    const response = await fetchJson(makeApiUrl("hst", urlParams), {
+        log:                 logFetches,
+        headers:             defaultHeaders,
+        sanitizeRequestLog:  {},
+        sanitizeResponseLog: {},
+    });
+
+    assertResponseSuccess(response, [
+        'OK',
+        'OK_SYNC',
+    ]);
+
+    return response.body;
+}
+
+/**
+ * @param response
+ * @param allowedStatuses
+ */
+const assertResponseSuccess = (response, allowedStatuses = ['OK']) => {
+    console.assert(
+        response.status === 200 && allowedStatuses.indexOf(response.body.status) > -1,
+        "non-successful response",
+        response
+    );
+};
+
+/**
+ * @param path
+ * @param params
+ * @returns {string}
+ */
+const makeApiUrl = (path, params = {}) => {
```

Comment:
there are already implemented libraries for making urls, as well as stuff in official API:

https://developer.mozilla.org/en-US/docs/Web/API/URL
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

I'm not sure they are available in mobile context, but they can be polyfilled
In any way, I'd strictly not recommend you to write your own url making utility library because you'd likely make the same mistakes people already did before you.

In your case (only 1 usage of search params generation) I'd go with formatting searchParams inline